holoviews/util/__init__.py

```diff
@@ -525,9 +536,17 @@ def _update_backend(cls, backend):
                         cls._create_builder(element, keywords))
 
         filtered_keywords = [k for k in all_keywords if k not in cls._no_completion]
-        kws = ', '.join('{opt}=None'.format(opt=opt) for opt in sorted(filtered_keywords))
-        old_doc = cls.__original_docstring__.replace('params(strict=Boolean, name=String)','')
-        cls.__doc__ = '\n    opts({kws})'.format(kws=kws) + old_doc
+        sorted_kw_set = sorted(set(filtered_keywords))
+        if sys.version_info.major == 2:
+            kws = ', '.join('{opt}=None'.format(opt=opt) for opt in sorted_kw_set)
+            old_doc = cls.__original_docstring__.replace(
+                'params(strict=Boolean, name=String)','')
+            cls.__doc__ = '\n    opts({kws})'.format(kws=kws) + old_doc
+        else:
+            signature = Signature([Parameter('args', Parameter.VAR_POSITIONAL)]
```

Comment:
No `self` needed in this case?

Reply:
Good point! I went back and checked and it seems the tab completion stopped working due to the lack of the positional arguments *other* than `cls`/`self`. In other words, if there are positional arguments and the signature doesn't specify any, the tab completion stops working. I've now simplified things in dc3c44401.

Reply:
So I'm confused, `cls` and `self` are or aren't needed?

Reply:
I get it now, nevermind.